First simple version of `flutter attach`.

floitschG opened the pull request.

floitschG committed:
First simple version of `flutter attach`.

floitschG committed:
First simple version of `flutter attach`.

mravn-google commented on packages/flutter_tools/lib/src/commands/attach.dart:
```diff
@@ -0,0 +1,78 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
```
2018 now

mravn-google commented on packages/flutter_tools/lib/executable.dart:
```diff
@@ -48,6 +49,7 @@ Future<Null> main(List<String> args) async {
 
   await runner.run(args, <FlutterCommand>[
     new AnalyzeCommand(verboseHelp: verboseHelp),
+    new AttachCommand(),
```
We should pass in `verboseHelp`

mravn-google commented on packages/flutter_tools/lib/src/commands/attach.dart:
```diff
@@ -0,0 +1,78 @@
+    ]; // observatoryUris;
+    final HotRunner hotRunner = new HotRunner(
+      <FlutterDevice>[flutterDevice],
+      debuggingOptions: new DebuggingOptions.enabled(getBuildInfo()),
```
Not really clear to me what the `BuildInfo` is used for in this context.

mravn-google reviewed (commented):
Just some superficial feedback, haven't tried to use this yet.

mravn-google committed:
Update all packages (#18506)

Hixie commented:
this looks great, thanks for doing this!

we should add a device lab integration test for this

cc @tvolkert

goderbauer committed:
Keep TextFields visible when keyboard comes up (#18291)

Fixes #10826.

Also in this PR: If you focus a text field, scroll it out of view and then start typing it will scroll back into view.

goderbauer committed:
Fix clipping for SliverLists (#18410)

Commit:
invert children for real (#18489)

Commit:
mention Text.rich on RichText (#18434)

sbaranov committed:
Add flutter tool support for creating app-specific VM snapshots. (#18417)

Add flutter tool support for creating app-specific VM snapshots.

Commit:
update Tristate checkbox semantics to consider indeterminate as "unchecked" (#18297)

Commit:
Fixed some typos (#18516)

liyuqian committed:
Skip painting hidden children of a sliver list (#18465)

We have many hidden children because of iOS accessibility (see also #18410).

For example, in the tile screen of complex_layout app, there could be up to 20 children being painted without this PR. With this PR, only 12 get painted.

This should improve the performance of FL-53 by 40%.

Hixie committed:
Remove race conditions involving finding available ports (#18488)

Hixie committed:
Revert "Remove race conditions involving finding available ports (#18488)" (#18521)

This reverts commit 775087225141573d20224e3cda3cff9e56a6cd1e.

goderbauer committed:
Fix aot_snapshot_size_* benchmarks (#18520)

... and make sure they fail if they can't report the size.

Commit:
Add minimum height and width box constraints to input decoration icons (#18480)

Commit:
Add semantics label property to Text widget (#18518)

goderbauer committed:
Add size benchmark for hello_world (#18522)

Commit:
manual rollback of of 18530 (#18531)

DanTup committed:
Don't be case-sensitive when checking for extension (#18515)

Seems like VS Code may have started using lowercase extension folders recently (our identifier is `Dart-Code.flutter` but now the folder is named `dart-code.flutter`), so this makes the check not sensitive to casing.

Also reuse extension identifier for download link

floitschG committed:
Forward port automatically.

floitschG committed:
Forward port automatically.

floitschG committed:
Roll sync async engine (#18454)

* Roll the engine.

This engine includes a VM that defaults to sync-async.

flutter/engine@c8eeee4 Roll the Dart VM.

floitschG committed:
Watching log for 'Observatory listening' to find port.

floitschG committed:
Watching log for 'Observatory listening' to find port.

gspencergoog committed:
Adding HSLColor and color 'within' matchers for HSVColor and HSLColor (#18294)

This adds an HSLColor class which uses a perceptual color space based upon human perception of colored light (as opposed to HSV, which is based on pigment colors).

You can see the difference in the color spaces here: https://en.wikipedia.org/wiki/HSL_and_HSV

I also added a "within" matcher for both HSLColor and HSVColor that will check if the (floating point) color components are within a certain error.

And tests.

goderbauer committed:
Report compressed/uncompressed size of icudtl, snapshots, and libflutter (#18583)

Reported for the hello_world compile benchmark in devicelab for now.

matthew-carroll committed:
Add docs to Opacity recommending AnimatedOpacity for opacity animatio… (#18092)

* Add docs to Opacity recommending AnimatedOpacity for opacity animations (#15464)

gspencergoog committed:
Roll engine to 0c119932c0d6cb30dafd75ec717d1eda76fd7651 (#18575)

floitschG committed:
Fix test (#18603)

Fix test.

This makes the buildbot green again.

awdavies committed:
[frdb] Added env test (#18485)

Hixie committed:
Revert matcher package (#18614)

The new matcher package deprecates isInstanceOf which seems eggregious.

matthew-carroll committed:
Add doc example to addLocalHistoryEntry in LocalHistoryRoute (#9392) (#18124)

* Add doc example to addLocalHistoryEntry in LocalHistoryRoute (#9392)

floitschG committed:
Make `ensureVersionFile` a `Future<void>` function. (#18631)

mravn-google committed:
Make Gradle script support Flutter-as-lib (#18633)

devoncarew committed:
revert some type args added in https://github.com/flutter/flutter/pull/18575 (#18621)

floitschG committed:
Wip.

floitschG committed:
Add test.

floitschG committed:
Merge branch 'flutter_attach' of github.com:floitschG/flutter into flutter_attach

floitschG commented:
Somehow made this PR unusable.
Creating a new one...
